src/app/core/basic-datatypes/string/edit-url/edit-url.component.ts

```diff
@@ -0,0 +1,52 @@
+import { Component } from "@angular/core";
+import { EditComponent } from "../../../entity/default-datatype/edit-component";
+import { DynamicComponent } from "../../../config/dynamic-components/dynamic-component.decorator";
+import { MatFormFieldModule } from "@angular/material/form-field";
+import { ReactiveFormsModule } from "@angular/forms";
+import { MatInputModule } from "@angular/material/input";
+import { ErrorHintComponent } from "../../../common-components/error-hint/error-hint.component";
+import { NgClass, NgIf } from "@angular/common";
+
+@DynamicComponent("EditUrl")
+@Component({
+  selector: "app-edit-url",
+  templateUrl: "./edit-url.component.html",
+  styleUrls: ["./edit-url.component.scss"],
+  imports: [
+    MatFormFieldModule,
+    ReactiveFormsModule,
+    MatInputModule,
+    ErrorHintComponent,
+    NgIf,
+    NgClass,
+  ],
+  standalone: true,
+})
+export class EditUrlComponent extends EditComponent<string> {
+  
+  /**
+   * Checks if the current formControl value is a valid URL.
+   * Returns true only if it is a properly formatted http/https URL.
+   */
+  isValidUrl(value: string | null): boolean {
```

Comment:
unit test cases to implement for this functionality:
- `it("should be invalid for value that is not a complete URL"`:  input value: `"example"`, expected result: `false`
- `it("should be valid for empty value"`:  input value: `""` or input value: `null` (test both), expected result: `true`
- `it("should be valid for url-like value"`:  input value: `"https://example.com"`, expected result: `true`